lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -97,186 +348,315 @@ def reference_time(cube, field):
         )
 
         cube.add_aux_coord(ref_time_coord)
 
 
-def experiment(cube, field):
-    """Add an 'experiment number' to the cube, if present in the field."""
-    if field.experiment_num != field.int_mdi:
-        cube.add_aux_coord(
-            DimCoord(field.experiment_num, long_name="experiment_number")
+def forecast_period(cube):
+    """
+    Add a forecast_period coord based on existing time and
+    forecast_reference_time coords.
+    """
+    try:
+        time_coord = cube.coord("time")
+        frt_coord = cube.coord("forecast_reference_time")
+    except CoordinateNotFoundError:
+        return
+    if len(time_coord.points) != 1 or len(frt_coord.points) != 1:
+        raise TranslationError(
+            "Unexpected number of points on time coordinates. Expected time:1; "
+            f"forecast_reference_time:1. Got {len(time_coord.points)}; "
+            f"{len(frt_coord.points)}"
         )
+    time_delta = time_coord.cell(0).point - frt_coord.cell(0).point
 
+    points = np.array(
+        time_delta.days * 24 * 60 * 60 + time_delta.seconds, dtype=np.int32
+    )
+    forecast_period_unit = cf_units.Unit("second")
+    if cube.coord("time").has_bounds():
+        time_window = time_coord.cell(0).bound
+        time_window = time_window[1] - time_window[0]
+        bounds = np.array(
+            [points - time_window.total_seconds(), points], dtype=np.int32
+        )
+    else:
+        bounds = None
+    cube.add_aux_coord(
+        iris.coords.AuxCoord(
+            points,
+            standard_name="forecast_period",
+            bounds=bounds,
+            units=forecast_period_unit,
+        )
+    )
 
-def proj_biaxial_ellipsoid(cube, field):
-    """
-    Ellipsoid definition is currently ignored.
 
+def mask_cube(cube, field):
     """
-    pass
-
+    Updates cube.data to be a masked array if appropriate.
 
-def tm_meridian_scaling(cube, field):
     """
-    Deal with the scale factor on the central meridian for transverse mercator
-    projections if present in the field.
 
-    Currently only caters for British National Grid.
-
-    """
-    if field.tm_meridian_scaling not in [field.float32_mdi, NIMROD_DEFAULT]:
-        if abs(field.tm_meridian_scaling - MERIDIAN_SCALING_BNG) < 1e-6:
-            pass  # This is the expected value for British National Grid
-        else:
-            warnings.warn(
-                "tm_meridian_scaling not yet handled: {}"
-                "".format(field.tm_meridian_scaling),
-                TranslationWarning,
+    if field.datum_type == 1:
+        # field.data are integers
+        if np.any(field.data == field.int_mdi):
+            cube.data = np.ma.masked_equal(cube.data, field.int_mdi)
+    elif field.datum_type == 0:
+        # field.data are floats
+        if np.any(np.isclose(field.data, field.float32_mdi)):
+            cube.data = np.ma.masked_inside(
+                cube.data, field.float32_mdi - 0.5, field.float32_mdi + 0.5
```

Comment:
It seems odd to re-judge here which are masked points, having just used "isclose" to see if there are any.
And I'm not confident of sufficient precision using +/-0.5 if float32_mdi can be quite large (?? practically, I don't know)
Suggest instead:
```
masked_points = np.isclose(field.data, field.float32_mdi)
if np.any(masked_points):
    cube.data = np.ma.masked_array(cube.data, mask=masked_points)
```

Reply:
Agreed. I've rearranged the integer bit to match this too and refactored appropriately.

Reply:
Very neat ! :+1: